x-pack/plugins/observability/public/hooks/slo/use_reset_slo.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useMutation } from '@tanstack/react-query';
+import { i18n } from '@kbn/i18n';
+import { IHttpFetchError, ResponseErrorBody } from '@kbn/core/public';
+import { useKibana } from '../../utils/kibana_react';
+
+type ServerError = IHttpFetchError<ResponseErrorBody>;
+
+export function useResetSlo() {
+  const {
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+  return useMutation<string, ServerError, { id: string; name: string }>(
+    ['resetSlo'],
+    ({ id }) => {
+      try {
+        return http.post(`/api/observability/slos/${id}/_reset`);
+      } catch (error) {
+        return Promise.reject(`Something went wrong: ${String(error)}`);
```

Comment:
i18n? 

Also, I would include `failed to reset {name}` in this error message as well.